### ISSUE TITLE
**** Major Blocker *****, Please fix immediately

### ISSUE THREAD

**Issue by krsolutions** [KasemJaffer/receive_sharing_intent#103]
e: /Users/reddy/flutter/.pub-cache/hosted/pub.dartlang.org/receive_sharing_intent-1.4.1/android/src/main/kotlin/com/kasem/receive_sharing_intent/ReceiveSharingIntentPlugin.kt: (139, 67): Type mismatch: inferred type is Uri? but Uri was expected
e: /Users/reddy/flutter/.pub-cache/hosted/pub.dartlang.org/receive_sharing_intent-1.4.1/android/src/main/kotlin/com/kasem/receive_sharing_intent/ReceiveSharingIntentPlugin.kt: (199, 95): Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type String?

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':receive_sharing_intent:compileDebugKotlin'.
> Compilation error. See log for more details

**Comment by themadmrj** [comment]
Still an issue as of today

**Comment by themadmrj** [comment]
Still an issue in the latest v1.4.3

**Comment by LazyDave76** [comment]
i've fired in a PR to try and fix this issue (https://github.com/KasemJaffer/receive_sharing_intent/pull/137)

try adding this to your pubspec.yaml file...
```
receive_sharing_intent:
    git:
      url: https://github.com/LazyDave76/receive_sharing_intent.git
      ref: 7ec782f
```

**Comment by themadmrj** [comment]
> i've fired in a PR to try and fix this issue (#137)
> 
> try adding this to your pubspec.yaml file...
> 
> ```
> receive_sharing_intent:
>     git:
>       url: https://github.com/LazyDave76/receive_sharing_intent.git
>       ref: 7ec782f
> ```

This works! Nice job! Did you also merge the pending PRs?

**Comment by LazyDave76** [comment]
> > i've fired in a PR to try and fix this issue (#137)
> > try adding this to your pubspec.yaml file...
> > ```
> > receive_sharing_intent:
> >     git:
> >       url: https://github.com/LazyDave76/receive_sharing_intent.git
> >       ref: 7ec782f
> > ```
> 
> This works! Nice job! Did you also merge the pending PRs?

no, just the failing on android issue

**Comment by themadmrj** [comment]
> > > i've fired in a PR to try and fix this issue (#137)
> > > try adding this to your pubspec.yaml file...
> > > ```
> > > receive_sharing_intent:
> > >     git:
> > >       url: https://github.com/LazyDave76/receive_sharing_intent.git
> > >       ref: 7ec782f
> > > ```
> > 
> > 
> > This works! Nice job! Did you also merge the pending PRs?
> 
> no, just the failing on android issue

Would be awesome if someone could maintain a new package, even just merging all the PRs would be good I think!

**Comment by LazyDave76** [comment]
> > > > i've fired in a PR to try and fix this issue (#137)
> > > > try adding this to your pubspec.yaml file...
> > > > ```
> > > > receive_sharing_intent:
> > > >     git:
> > > >       url: https://github.com/LazyDave76/receive_sharing_intent.git
> > > >       ref: 7ec782f
> > > > ```
> > > 
> > > 
> > > This works! Nice job! Did you also merge the pending PRs?
> > 
> > 
> > no, just the failing on android issue
> 
> Would be awesome if someone could maintain a new package, even just merging all the PRs would be good I think!

@KasemJaffer will get around to it, I'm sure :)

**Comment by themadmrj** [comment]
> > > > > i've fired in a PR to try and fix this issue (#137)
> > > > > try adding this to your pubspec.yaml file...
> > > > > ```
> > > > > receive_sharing_intent:
> > > > >     git:
> > > > >       url: https://github.com/LazyDave76/receive_sharing_intent.git
> > > > >       ref: 7ec782f
> > > > > ```
> > > > 
> > > > 
> > > > This works! Nice job! Did you also merge the pending PRs?
> > > 
> > > 
> > > no, just the failing on android issue
> > 
> > 
> > Would be awesome if someone could maintain a new package, even just merging all the PRs would be good I think!
> 
> @KasemJaffer will get around to it, I'm sure :)

When fetching with your git repo, the build fails on Xcode `Module receive_sharing_intent not found`. Does it work for you?

**Comment by LazyDave76** [comment]
> > > > > > i've fired in a PR to try and fix this issue (#137)
> 
> > > > > > try adding this to your pubspec.yaml file...
> 
> > > > > > ```
> 
> > > > > > receive_sharing_intent:
> 
> > > > > >     git:
> 
> > > > > >       url: https://github.com/LazyDave76/receive_sharing_intent.git
> 
> > > > > >       ref: 7ec782f
> 
> > > > > > ```
> 
> > > > > 
> 
> > > > > 
> 
> > > > > This works! Nice job! Did you also merge the pending PRs?
> 
> > > > 
> 
> > > > 
> 
> > > > no, just the failing on android issue
> 
> > > 
> 
> > > 
> 
> > > Would be awesome if someone could maintain a new package, even just merging all the PRs would be good I think!
> 
> > 
> 
> > @KasemJaffer will get around to it, I'm sure :)
> 
> 
> 
> When fetching with your git repo, the build fails on Xcode `Module receive_sharing_intent not found`. Does it work for you?

Yeah, builds fine for me on Xcode with flutter 1.22.6 - maybe try running 

Flutter clean

And trying again?
